Mathlib/Algebra/DirectSum/Module.lean

```diff
@@ -453,6 +453,13 @@ theorem isInternal_submodule_iff_isCompl (A : ι → Submodule R M) {i j : ι} (
   exact ⟨fun ⟨hd, ht⟩ ↦ ⟨hd, codisjoint_iff.mpr ht⟩, fun ⟨hd, ht⟩ ↦ ⟨hd, ht.eq_top⟩⟩
 #align direct_sum.is_internal_submodule_iff_is_compl DirectSum.isInternal_submodule_iff_isCompl
 
+-- Controversial lemma (EricW not a fan). Let's use to get sorry-free and then reconsider
```

Comment:
This lemma was excised from a previous PR #8369

@eric-wieser raised some good points about how we should prove this lemma [here](https://github.com/leanprover-community/mathlib4/pull/8369#discussion_r1390509279)

Reply:
Now that this PR compiles, what is the status here?

Reply:
I have removed this code comment in the commit 4f9a599de5a6ded95c67869481e586287e48c61e

I believe Eric's position was that it might be worth proving this lemma without using `DirectSum.isInternal_submodule_iff_independent_and_iSup_eq_top` so that we would be able to apply it to additive monoids. I feel this is a large detour considering we have no application in mind and this is only a two-line proof. However I recognise his position is reasonable.

Reply:
Ok, then I'm going to merge. We can refactor later if needed.